chain-impl-mockchain/tests/ledger/discrimination_tests.rs

```diff
@@ -0,0 +1,115 @@
+use crate::common::address::AddressData;
+use crate::common::ledger;
+use crate::common::ledger::ConfigBuilder;
+use crate::common::tx_builder::TransactionBuilder;
+use chain_addr::{Discrimination, Kind, KindType};
+use chain_impl_mockchain::transaction::*;
+use chain_impl_mockchain::value::*;
+use quickcheck::{Arbitrary, Gen, TestResult};
+use quickcheck_macros::quickcheck;
+
+#[derive(Clone, Debug)]
+pub struct ArbitraryAddressKind(KindType);
+
+impl Arbitrary for ArbitraryAddressKind {
```

Comment:
There is already an `Arbitrary` impl for `KindType`, conditionally enabled on the `chain-addr` crate with feature `property-test-api`. Consider using that instead of a shim type?

Reply:
Yes, but this property is throwing `not implement yet` exception for `Multisig` value, so i created another without messing value